apps/mobile/src/features/browser/approver-sheet/stx/call-contract/hooks.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback } from 'react';
+
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { bytesToHex } from '@stacks/common';
+
+import { RpcRequest, stxCallContract } from '@leather.io/rpc';
+import {
+  TransactionTypes,
+  generateStacksUnsignedTransaction,
+  getStacksAssetStringParts,
+} from '@leather.io/stacks';
+import { useOnMount } from '@leather.io/ui/native';
+
+import { StxNormalizedParams } from '../utils';
+
+interface UseCallContractTxHex {
+  message: RpcRequest<typeof stxCallContract>;
+  normalizedParams: StxNormalizedParams;
+  setTxHex(txHex: string): void;
+}
+
+export function useCallContractTxHex({
+  message,
+  normalizedParams,
+  setTxHex,
+}: UseCallContractTxHex) {
+  const { fromAccountId } = useStacksSigners();
+
+  const getTxHex = useCallback(
+    async function getTxHex() {
+      const signer = fromAccountId(normalizedParams.accountId)[0];
+      if (!signer?.publicKey) throw new Error('yeet');
+      const { contractAddress, contractName } = getStacksAssetStringParts(message.params.contract);
+      const tx = await generateStacksUnsignedTransaction({
+        ...normalizedParams,
+        txType: TransactionTypes.ContractCall,
+        contractAddress,
+        contractName,
+        functionName: message.params.functionName,
+        functionArgs: message.params.functionArgs ?? [],
+        publicKey: bytesToHex(signer.publicKey),
+      });
+      const txHex = tx.serialize();
+      return txHex;
+    },
+    [
+      fromAccountId,
+      message.params.contract,
+      message.params.functionName,
+      message.params.functionArgs,
+      normalizedParams,
+    ]
+  );
+  useOnMount(() => {
+    void getTxHex().then(txHex => setTxHex(txHex));
+  });
```

Comment:
Why only on mount and not when deps change?

Reply:
Similar to your other PR we need to check if setting the `txHex` in state is the best way